src/models/room.ts

```diff
@@ -1963,6 +1971,8 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
     private async fetchRoomThreadList(filter?: ThreadFilterType): Promise<void> {
         const timelineSet = filter === ThreadFilterType.My ? this.threadsTimelineSets[1] : this.threadsTimelineSets[0];
 
+        if (!timelineSet) return;
+
```

Comment:
Following from the discussion recently about implicit casting to boolean, I have a question and also I'm wondering if this early return could be even earlier.

My thinking is along the lines of:

- this check is I think saying "if `timelineSet` is undefined, don't go any further" (I don't know if we should also stop if the array/set is empty, should we continue in that case?)
- if we're only concerned about undefined, the types would suggest that either both entries in `this.threadsTimelineSets` are defined or undefined at the same time
- in which case, could we just check if `this.timelineSet` has a length of 0 before assigning `timelineSet` to do the early return?

Reply:
Thanks for the review. That makes sense :+1:

Reply:
:thinking: if checking for the length it would need some non-null assertions :neutral_face: 

```
        if (this.createThreadsTimelineSets.length === 0) return;

        const timelineSet = filter === ThreadFilterType.My ? this.threadsTimelineSets[1]! : this.threadsTimelineSets[0]!;
```

I personally prefer to avoid them because they short-circuit the type system by knowing better based on knowledge.

Reply:
Perhaps I haven't got the correct settings turned on in TS config, but for me when I try that code it seems to allow TS to correctly infer that the type has been narrowed to the two item array (and so doesn't need the non-null assertion).

I definitely agree about trying to avoid the `!` though.